lib/statesman/adapters/active_record_queries.rb

```diff
@@ -30,6 +34,30 @@ def most_recent_transition_join
 
         private
 
+        def in_state_without_initial(*states)
+          states = states.flatten.map(&:to_s)
+
+          joins(most_recent_transition_join_without_initial).
+            where(states_where(most_recent_transition_alias, states), states).
```

Comment:
It feels a bit weird to me that we are still using the `to_state IS NOT NULL` condition despite using `JOIN` which guarantees that not to be the case. Unless someone is using `to_state = NULL` on some records which honestly I don't think we should support.

Reply:
Hmm, interesting point. If we don't want to support `to_state = NULL`, we should make that (technically breaking) change separately, right?